### ISSUE TITLE
Incorrect parsed name from Google Maps api

### ISSUE THREAD

**Issue by jongotlin** [geocoder-php/Geocoder#1077]
Searching for __Pershagen__  returns the following json
https://maps.googleapis.com/maps/api/geocode/json?address=Pershagen&key=xxx
```json
{
   "results" : [
      {
         "address_components" : [
            {
               "long_name" : "Pershagen",
               "short_name" : "Pershagen",
               "types" : [ "locality", "political" ]
            },
            {
               "long_name" : "Södertälje",
               "short_name" : "Södertälje",
               "types" : [ "postal_town" ]
            },
            {
               "long_name" : "Stockholm County",
               "short_name" : "Stockholm County",
               "types" : [ "administrative_area_level_1", "political" ]
            },
            {
               "long_name" : "Sweden",
               "short_name" : "SE",
               "types" : [ "country", "political" ]
            },
            {
               "long_name" : "151 39",
               "short_name" : "151 39",
               "types" : [ "postal_code" ]
            }
         ],
         "formatted_address" : "151 39 Pershagen, Sweden",
         "geometry" : {
            "location" : {
               "lat" : 59.14664239999999,
               "lng" : 17.6511284
            },
            "location_type" : "APPROXIMATE",
            "viewport" : {
               "northeast" : {
                  "lat" : 59.15236419999999,
                  "lng" : 17.6671358
               },
               "southwest" : {
                  "lat" : 59.1409196,
                  "lng" : 17.635121
               }
            }
         },
         "place_id" : "ChIJV_Hb5dUSX0YRULaMsvX-AAo",
         "types" : [ "locality", "political" ]
      }
   ],
   "status" : "OK"
}
```

Using the google maps provider I get the following object. The phrase __Pershagen__ is omitted.
![image](https://user-images.githubusercontent.com/165154/90138748-c1c54880-dd77-11ea-8be5-8ec5279d56f2.png)

Searching for the similar area __Ronna__ I get a more correct object.
![image](https://user-images.githubusercontent.com/165154/90139396-98f18300-dd78-11ea-89d9-67ad14e2d05f.png)

When looping over address components __Pershagen__ is first set but then overridden by __Södertälje__. 

Reversing the array on L235 in GoogleMaps.php returns a more accurate result.
```diff
+foreach (array_reverse($result->address_components) as $component) {
-foreach ($result->address_components as $component) {
```
![image](https://user-images.githubusercontent.com/165154/90141968-166ac280-dd7c-11ea-9093-59969cacd5d7.png)

Not at all sure how this will affect other results!

**Comment by alanpoulain** [comment]
It's because of the order of these two lines:
https://github.com/geocoder-php/Geocoder/blob/c2d2c3bc9dde6ee92954cb2f427e0e1c866a385d/src/Provider/GoogleMaps/GoogleMaps.php#L295-L296
Maybe `postal_town` should be before `locality`?

**Comment by jongotlin** [comment]
@alanpoulain That won't affect the result unfortunately.

**Comment by jbelien** [comment]
We'll need to test for address all over the world to (maybe) improve the process of `locality` and `postal_town`.
I'll try to do that ASAP.

**Comment by jongotlin** [comment]
Ideally Pershagen would be a sub locality and Södertälje locality but thats not what the Google response says. 

Can we have Södertälje in list of admin levels?

**Comment by jbelien** [comment]
> Can we have Södertälje in list of admin levels?

The issue is to keep a logic that works for all countries.
We only put `administrative_area_level_1...5` in the admin levels list.

**Comment by jongotlin** [comment]
Aha, understand. Maybe skip Södertälje in this case. The data from Google is not really correct. I think the rule here should be "If locality is set it should not be overridden by postal_town".

**Comment by jbelien** [comment]
The goal of this library is to create a layer on top of (reverse) geocoding API, not fix/process/mess with the data returned by the API.

> I think the rule here should be "If locality is set it should not be overridden by postal_town".

I'll make some tests for other countries in the world to see if that applies.
We won't make changes that will affect every country "just" to fix a locality issue in Sweden.

**Comment by jongotlin** [comment]
Yes, understand and agree with that. However, overriding a value like this doesn't seems correct.
